src/main/java/gregtech/api/net/CPacketMultiBlockStructure.java

```diff
@@ -0,0 +1,141 @@
+package gregtech.api.net;
+
+import gregtech.api.GregTechAPI;
+import gregtech.api.block.machines.MachineItemBlock;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.api.metatileentity.multiblock.IMultiblockAbilityPart;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.ItemBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CPacketMultiBlockStructure implements NetworkHandler.Packet {
+    public List<ItemStack> map;
+    public List<BlockInfo> blockInfos;
+    public int dim;
+
+    public CPacketMultiBlockStructure(List<ItemStack> map, List<BlockInfo> blockInfos, int dim) {
+        this.map = map;
+        this.blockInfos = blockInfos;
+        this.dim = dim;
+    }
+
+    public static void registerPacket(int packetId) {
+        NetworkHandler.registerPacket(packetId, CPacketMultiBlockStructure.class, new NetworkHandler.PacketCodec<>(
+                (packet, buf) -> {
+                    buf.writeVarInt(packet.dim);
+                    buf.writeVarInt(packet.map.size());
+                    packet.map.forEach(buf::writeItemStack);
+                    buf.writeVarInt(packet.blockInfos.size());
+                    packet.blockInfos.forEach(blockInfo -> blockInfo.writeBuf(buf));
+                },
+                (buf) -> {
+                    int dim = buf.readVarInt();
+                    List<ItemStack> map = new ArrayList<>();
+                    List<BlockInfo> blockInfos = new ArrayList<>();
+                    int size = buf.readVarInt();
+                    try {
+                        for (int i = 0; i < size; i++) {
+                            map.add(buf.readItemStack());
+                        }
+                    } catch (Exception ignored) {
```

Comment:
Shouldn't there be some logging here instead of silently eating exceptions?
Or are these always expected for some reason producing too much noise?